packages/@react-aria/tag/src/useTag.ts

```diff
@@ -63,9 +63,20 @@ export function useTag<T>(props: AriaTagProps<T>, state: ListState<T>, ref: RefO
     if (e.key === 'Delete' || e.key === 'Backspace') {
       onRemove(item.key);
       e.preventDefault();
+
+      // Focus container if last tag was removed.
+      if (state.collection.size === 1 && state.collection.getFirstKey() === item.key) {
+        let lastTag = ref;
+        let container = ref.current.parentElement;
+        setTimeout(() => {
+          if (!lastTag.current) {
+            container.focus();
+          }
+        }, 50);
```

Comment:
why 50ms out of curiosity? why not immediately? I assume it's because the tag container doesn't get a tabIndex of -1 until the state change has taken effect? we could instead use a useEffect and a ref, the ref tracks previous count of tags and if the tag group has focus, if the count of tags goes from > 0 to === 0, then we focus the container

does focus get set to the container if we mouse click to remove the last tag already?

Reply:
The intention was to wait and check that the list tag actually got removed from the DOM. Mostly in order to avoid the case that it wasn't removed immediately (like a confirmation dialog before it actually gets removed). I think your suggestion is much better though, so I just changed it to that.